.github/workflows/build.yml

```diff
@@ -265,6 +265,20 @@ jobs:
           name: ${{ steps.diff.outputs.diff_name }}
           path: ${{ steps.diff.outputs.diff_name }}
 
+  aws-sdk-spec-test:
+    needs:
+      - build
+    if: github.event_name == 'push'
+    uses: ./.github/workflows/tf-aws-test.yml
+    secrets: inherit
+
+  file-diff-sdk-spec-test:
+    needs:
+      - build
+    if: github.event_name == 'pull_request'
+    uses: ./.github/workflows/wing-test.yml
+    secrets: inherit
```

Comment:
Both PR and main branch pushes can make use of a diff-based approach. Ideally turbo should be used to get a good diff. This use case is not well documented and I would be happy to meet on it.